components/tidb_query_expr/src/impl_string.rs

```diff
@@ -219,6 +221,80 @@ pub fn rpad(arg: BytesRef, len: &Int, pad: BytesRef, writer: BytesWriter) -> Res
     }
 }
 
+#[rpn_fn(writer, capture = [ctx])]
+#[inline]
+pub fn add_string_and_duration(
+    ctx: &mut EvalContext,
+    arg0: BytesRef,
+    arg1: &Duration,
+    writer: BytesWriter,
+) -> Result<BytesGuard> {
+    let duration = Duration::parse(ctx, arg0, MAX_FSP);
+    if duration.is_ok() {
+        let arg0 = duration.unwrap();
+        let dur = match arg0.checked_add(*arg1) {
+            Some(dur) => dur,
+            None => return Ok(writer.write(None)),
```

Comment:
Please use this handler when time error occurs:

https://github.com/tikv/tikv/blob/a1b329dc15cc599a0f916bb1bb912e43ef9e88e8/components/tidb_query_expr/src/impl_time.rs#L258-L259